internal/resources/business_unit/resource.go

```diff
@@ -0,0 +1,934 @@
+							Optional:    true,
+						},
+						"region": schema.StringAttribute{
+							Description: "Name of the region",
+							Optional:    true,
+						},
+						"state": schema.StringAttribute{
+							Description: "Name of the state",
+							Optional:    true,
+						},
+						"company": schema.StringAttribute{
+							Description: "Name of the company",
+							Optional:    true,
+						},
+						"department": schema.StringAttribute{
+							Description: "Name of the department",
+							Optional:    true,
+						},
+						"building": schema.StringAttribute{
+							Description: "Name or number of the building",
+							Optional:    true,
+						},
+						"apartment": schema.StringAttribute{
+							Description: "Name or number of the apartment",
+							Optional:    true,
+						},
+						"po_box": schema.StringAttribute{
+							Description: "Post office box number",
+							Optional:    true,
+						},
+						"phone": schema.StringAttribute{
+							Description: "Phone number",
+							Optional:    true,
+						},
+						"mobile": schema.StringAttribute{
+							Description: "Mobile phone number",
+							Optional:    true,
+						},
+						"email": schema.StringAttribute{
+							Description: "Email address",
+							Optional:    true,
+						},
+						"fax": schema.StringAttribute{
+							Description: "Fax number",
+							Optional:    true,
+						},
+						"additional_address_info": schema.StringAttribute{
+							Description: "Further information on the Address",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"parent_unit": schema.ListNestedBlock{
+				Description: "Reference to a parent Business Unit by its key.",
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.SizeAtLeast(1),
+					listvalidator.IsRequired(),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Description: "User-defined unique identifier of the Business Unit",
+							Required:    true,
+						},
+						"type_id": schema.StringAttribute{
+							Description: "The type of the Business Unit",
+							Required:    true,
+							Validators: []validator.String{
+								stringvalidator.OneOf(CompanyType, DivisionType),
+							},
+						},
+					},
+				},
+			},
+			"stores": schema.ListNestedBlock{
+				Description: "Stores that are part of the Business Unit.",
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Description: "User-defined unique identifier of the Store",
+							Required:    true,
+						},
+						"type_id": schema.StringAttribute{
+							Description: "The type of the Store",
+							Required:    true,
+							Validators: []validator.String{
+								stringvalidator.OneOf(StoreTypeID),
+							},
+						},
+					},
+				},
+			},
+			"inherited_associates": schema.ListNestedBlock{
```

Comment:
I don't think this one will have much value in the context of a terraform provider (it is eventually consistent and a result of other resources made). Do you have a usecase where this might be useful?